library/pod-security-policy/selinux/src_test.rego

```diff
@@ -0,0 +1,150 @@
+    results := violation with input as input
+    count(results) == 0
+}
+
+test_input_seLinux_option_not_allowed_not_in_list {
+    input := { "review": input_review, "parameters": input_parameters_not_in_list}
+    results := violation with input as input
+    count(results) > 0
+}
+
+test_input_seLinux_options_no_security_context {
+    input := { "review": input_review_no_security_context, "parameters": input_parameters_in_list}
+    results := violation with input as input
+    count(results) == 0
+}
+
+test_input_seLinux_options_two_allowed_in_list {
+    input := { "review": input_review_two, "parameters": input_parameters_in_list}
+    results := violation with input as input
+    count(results) == 0
+}
+
+test_input_seLinux_option_two_not_allowed_not_in_list {
+    input := { "review": input_review_two, "parameters": input_parameters_not_in_list}
+    results := violation with input as input
+    count(results) > 0
+}
+
+test_input_seLinux_options_many_allowed_in_list {
+    input := { "review": input_review_many, "parameters": input_parameters_in_list}
+    results := violation with input as input
+    count(results) == 0
+}
+
+test_input_seLinux_options_many_not_allowed_not_in_list {
+    input := { "review": input_review_many, "parameters": input_parameters_not_in_list}
+    results := violation with input as input
+    count(results) > 0
+}
+
+test_input_seLinux_options_many_not_allowed_not_in_list_two {
+    input := { "review": input_review_many, "parameters": input_parameters_not_in_list_two}
+    results := violation with input as input
+    count(results) > 0
+}
+
+input_review = {
+    "object": {
+        "metadata": {
+            "name": "nginx"
+        },
+        "spec": {
+            "containers": input_containers_one,
+      }
+    }
+}
+
+input_review_two = {
+    "object": {
+        "metadata": {
+            "name": "nginx"
+        },
+        "spec": {
+            "securityContext": input_seLinuxOptions
+        }
+    }
+}
+
+input_review_no_security_context = {
+    "object": {
+        "metadata": {
+            "name": "nginx"
+        },
+        "spec": {}
+    }
+}
+
+input_review_many = {
+    "object": {
+        "metadata": {
+            "name": "nginx"
+        },
+        "spec": {
+            "containers": input_containers_many,
+      }
+    }
+}
+
+input_containers_one = [
+{
+    "name": "nginx",
+    "image": "nginx",
+    "securityContext": input_seLinuxOptions
+}]
+
+input_containers_many = [
+{
+    "name": "nginx",
+    "image": "nginx",
+    "securityContext": {
+        "privileged": true
+    }
+},
+{
+    "name": "nginx2",
+    "image": "nginx",
+},
+{
+    "name": "nginx3",
+    "image": "nginx",
+    "securityContext": input_seLinuxOptions
+}]
+
+input_seLinuxOptions = {
+    "seLinuxOptions": {
+        "level": "s0:c123,c456",
+        "role": "object_r",
+        "type": "svirt_sandbox_file_t",
+        "user": "system_u"
+    }
```

Comment:
add a test for when only a subset of `allowedSELinuxOptions` is provided in `seLinuxOptions`
and another test for when only a subset of `seLinuxOptions` is in `allowedSELinuxOptions`

Reply:
thanks, added